app/src/main/java/com/group80/uoftinder/ContactsActivity.java

```diff
@@ -0,0 +1,134 @@
+
+        // TODO: remove after having functioning log-in ********************************************
+        FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+        firebaseAuth.signInWithEmailAndPassword("csc207.group80.uoftinder@gmail.com", "CSC207Group80!").addOnFailureListener(Throwable::printStackTrace);
+//        firebaseAuth.signInWithEmailAndPassword("csc207.group80.uoftinder.bot@gmail.com", "12345678").addOnFailureListener(Throwable::printStackTrace);
+
+        // TODO: wrap-up FirebaseFirestore
+        FirebaseFirestore firebaseFirestore = FirebaseFirestore.getInstance();
+        String uid = firebaseAuth.getCurrentUser().getUid();
+        Query query = firebaseFirestore.collection("Users").whereArrayContains("contacts", uid);
+        //******************************************************************************************
+
+        FirestoreRecyclerOptions<ContactModel> contacts = new FirestoreRecyclerOptions.Builder<ContactModel>().setQuery(query, ContactModel.class).build();
+        contactAdapter = new FirestoreRecyclerAdapter<ContactModel, ContactViewHolder>(contacts) {
+            @Override
+            protected void onBindViewHolder(@NonNull ContactViewHolder holder, int position, @NonNull ContactModel contactModel) {
+                holder.onBind(getApplicationContext(), position, contactModel);
+            }
+
+            @NonNull
+            @Override
+            public ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+                View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.uoftinder_contact_list_contact_layout, parent, false);
+                return new ContactViewHolder(view);
+            }
+        };
+
+        RecyclerView recyclerView = findViewById(R.id.contactActivityRecyclerView);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.setAdapter(contactAdapter);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        contactAdapter.startListening();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (contactAdapter != null) {
+            contactAdapter.stopListening();
+        }
+    }
+
+    private class ContactViewHolder extends RecyclerView.ViewHolder {
+        protected final TextView contactName;
+        protected final ImageView contactPic;
+
+        public ContactViewHolder(@NonNull View itemView) {
+            super(itemView);
+            contactName = itemView.findViewById(R.id.contactListContactName);
+            contactPic = itemView.findViewById(R.id.contactListContactPic);
+        }
+
+        /**
+         * Triggered when this View Holder is bind to an adapter
+         * <p>
+         * // TODO: this method may require architectural fix
```

Comment:
Just wondering what this means - could you be specific?

Reply:
* For line 97: think of `adapter` as a `List<ContactViewHolder>`. This `onBind` method is called automatically on an instance of the `ContactViewHolder` when it is added to the list. 
* Fore line 98: it's a tag for perserving white space in Javadoc.
* Fore line 99: i'm working on the transition to MVP (see [here](https://github.com/CSC207-2022F-UofT/course-project-group-80/pull/56#discussion_r1036732564)) and hopefullt this wouldn't be an issue after the transition. Sorry for the confusion.